### PR TITLE
update/SOF 7469

### PR DESCRIPTION
**Pull request by VsevolodX**
- update: fix oxi state that we don't use to 0
- update: add top level function to create multiple defects

### CODE REVIEW

File: src/py/mat3ra/made/tools/build/defect/__init__.py [hunk 1]
```diff
@@ -36,6 +36,33 @@ def create_defect(
     return builder.get_material(configuration) if builder else configuration.crystal
 
 
+def create_defects(
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/__init__.py:39]: We should create a test for this

File: src/py/mat3ra/made/tools/README.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+- `create_defect`
+- `create_slab_defect`
+- `create_slab`
+- `create_perturbation`
+- `create_passivation`
+- `create_nanoribbon`
+- `create_grain_boundary`
+
+Some of the top-level functions are helpers that allow to get additional properties from configuration to refine the build process, such as:
+- `get_termaintions` for slab
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/README.md:118]: get_terminations

File: src/py/mat3ra/made/tools/build/passivation/__init__.py [hunk 3]
```diff
@@ -16,3 +16,10 @@ def create_passivation(
     if builder is None:
         builder = SurfacePassivationBuilder(build_parameters=SurfacePassivationBuilderParameters())
     return builder.get_material(configuration)
+
+
+def get_unique_coordination_numbers(configuration: PassivationConfiguration) -> set:
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/passivation/__init__.py:21]: Should be in analyze

File: src/py/mat3ra/made/tools/README.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+- BuilderParameters: specific parameters of the digital representation of the transformations.
+- (Result) Materials: the output of the Builders (e.g. point defect in Silicon, graphene nanoribbon etc.).
+
+[//]: # (a diagram that explains the relationships between entities)
+![Made Tools](images/made-tools.png)
+
+for a specific case of the creation of an interface:
+
+[//]: # (a diagram that explains the relationships for an Interface)
+![Made Tools](images/made-tools.png)
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/README.md:80]: To be ignored in pyproject

File: src/py/mat3ra/made/tools/README.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+- `get_terminations` for slab
+- `get_unique_coordination_numbers` for passivation
+- `get_optimal_film_displacement` for interface
+
+
+## 3. Usage
+
+### 3.1. Installation
+
+See top-level [README](LINK) for installation instructions.
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/README.md:127]: Link

File: src/py/mat3ra/made/tools/README.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+## Links
+
+1. [Exabyte Source of Schemas and Examples (ESSE), Github Repository](https://github.com/exabyte-io/exabyte-esse)
+2. [ECMAScript 2015 Language Specifications](https://www.ecma-international.org/ecma-262/6.0/)
+3. [POSCAR file format, official website](https://cms.mpi.univie.ac.at/vasp/guide/node59.html)
+4. [XYZ file format, Wikipedia](https://en.wikipedia.org/wiki/XYZ_file_format)
+5. [Quantum ESPRESSO, Official Website](https://www.quantum-espresso.org/)
+6. [Mocha, Official Website](https://mochajs.org/)
+7. [Git LFS, Official Website](https://git-lfs.github.com/)
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/README.md:178]: We need to update these links
- ASE
- Pymatgen
- https://arxiv.org/abs/1902.10838, ESSE github

File: src/py/mat3ra/made/tools/build/defect/configuration.py [hunk 7]
```diff
@@ -54,20 +55,37 @@ def from_site_id(
...
+    def from_dict(cls, crystal: Material, data: dict):
+        if "site_id" in data:
+            config = PointDefectConfiguration.from_site_id(crystal=crystal, **data)
+        elif "coordinate" in data:
+            config = PointDefectConfiguration(crystal=crystal, **data)
+        elif "approximate_coordinate" in data:
+            config = PointDefectConfiguration.from_approximate_coordinate(crystal=crystal, **data)
+        else:
+            raise ValueError(f"Invalid defect configuration: {data}")
+        return config
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/configuration.py:87]: Constructor needs to return `cls()`. We can override the default constructor with it.

File: src/py/mat3ra/made/tools/README.md (hunk 4 above)
Comment (VsevolodX) [reply src/py/mat3ra/made/tools/README.md]: I'll place images at the top-level, since they probably can be reused later

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**